extension/src/types.ts

```diff
@@ -85,3 +85,64 @@ export enum MessageType {
     DEBUG_MESSAGE,
     CHANGE_ROOM,
 }
+
+/**
+ * Video status for one user
+ */
+enum UserVideoStatus {
+    Unknown, // Still Loading of DOM, or not on the video URL, or any othe reason (no info from Video API)
+    Buffering, // has not reached minBufferLength (video is paused or not started)
+    Playing, // is currently playing the video
+    Ready, // (equivalent of Waiting or Paused) has buffered enough of the video, is pending start signal
+}
+
+/**
+ * Video status for all players that have completed initial Sync
+ */
+enum VideoSyncStatus {
+    Paused, // video is paused and no action should be taken (apart from buffering)
+    Waiting, // waiting for all players to be in waiting status before playing
+    Playing, // video should playing normally on all user browsers
+}
+
+/**
+ * Information about one user
+ */
+export class User {
```

Comment:
I fell like it's not the right name.

Maybe UserSync ? ParticipantSync ?

Reply:
What about `Watcher` ?